jl4-core/src/L4/TypeCheck/Environment/TH.hs

```diff
@@ -10,14 +10,24 @@ import L4.Syntax as L4
 import L4.Lexer
 import L4.Parser.SrcSpan
 
-data Renamed = MkRenamed {orig :: String, chosen :: Maybe String}
+data Renamed = MkRenamed
+  { orig :: String
+  , chosen :: Maybe String
+  , variants :: Maybe [String]
+  -- ^ There cannot be more than 10 variants for the same name
+  -- at the moment.
+  -- This is an arbitrary cut-off, that is easily fixed.
```

Comment:
this is a really unnecessary hack - there's no actual need to do this. Honestly, I would prefer if you had just given each of the variants a name, like 

leqString, etc. 

if this is starting to become a problem, we might need to look for a different abstraction, but this is not nice...